### PR TITLE
Supermicro_arm64_1G_Switch_G3748_Added: Patch of dts file to support …

### PR DESCRIPTION
**Pull request by SelvaSupermicro**
This patch is to support new platform, Supermicro 1G arm64 switch G3748 which uses marvel AC5X ASIC.
This patch is applied for all arm64 builds and will be effective only for this new platform.  

Testing:
1) Tested running M0/PTF32 test suites for this new platform.
2) Verified image partitions  
3) Verified uboot and sonic platform functionalities

### CODE REVIEW

**Comment by saiarcot895** [comment]
Has/will this been upstreamed?

**Comment by SelvaSupermicro** [comment]
Hello Saikrishna
Yes, once the PR is reviewed and approved it should automatically get merged to master (upstream)
I am sure you know that process.So I really don’t completely understand your question.
Please help me understand I would be glad to respond.

Thank you
Selva.
Sent from my iPhone

On Jan 11, 2024, at 5:00 PM, Saikrishna Arcot ***@***.***> wrote:

﻿
[CAUTION: External Mail]

Has/will this been upstreamed?

File: patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch
Comment (paulmenzel) [inline patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch:1]: Please add patch files creates with `git format-patch` containing a suitable and elaborate commit message including Signed-off-by lines

**Review by paulmenzel — CHANGES_REQUESTED** [review]
Thank you for your contribution.

PS: If you reply via email, please remove the cited part, otherwise the comment in the GitHub Web interface is “ugly” to read.

**Comment by saiarcot895** [comment]
> Yes, once the PR is reviewed and approved it should automatically get merged to master (upstream)
I am sure you know that process.So I really don’t completely understand your question.
Please help me understand I would be glad to respond.

Hi @SelvaSupermicro,

By upstream, I mean submitting this change into the main Linux kernel. See [their contribution page](https://docs.kernel.org/process/submitting-patches.html) for more details.

**Comment by SelvaSupermicro** [comment]
> > Yes, once the PR is reviewed and approved it should automatically get merged to master (upstream)
> > I am sure you know that process.So I really don’t completely understand your question.
> > Please help me understand I would be glad to respond.
> 
> Hi @SelvaSupermicro,
> 
> By upstream, I mean submitting this change into the main Linux kernel. See [their contribution page](https://docs.kernel.org/process/submitting-patches.html) for more details.

Thanks for the clarifications. These code changes are applicable only for SONiC. Hence we dont want to  upstream to main Linux kernel.

**Comment by paulmenzel** [comment]
From your commit message:

> hence will not be upstreamed to main ONL repo

SONiC and ONL are two separate projects for a network operating system. No idea, why you reference ONL.

To ease maintenance of the SONiC Linux kernel repository, you have to sent these patches upstream. Often upstream is also giving you a high quality review, that the SONiC developers/maintainers are not capable of.

Also the devicetree has a lot of commented out lines, that also need to be cleaned up.

**Comment by SelvaSupermicro** [comment]
> From your commit message:
> 
> > hence will not be upstreamed to main ONL repo
> 
> SONiC and ONL are two separate projects for a network operating system. No idea, why you reference ONL.
> 
> To ease maintenance of the SONiC Linux kernel repository, you have to sent these patches upstream. Often upstream is also giving you a high quality review, that the SONiC developers/maintainers are not capable of.
> 
> Also the devicetree has a lot of commented out lines, that also need to be cleaned up.



> From your commit message:
> 
> > hence will not be upstreamed to main ONL repo
> 
> SONiC and ONL are two separate projects for a network operating system. No idea, why you reference ONL.
> 
> To ease maintenance of the SONiC Linux kernel repository, you have to sent these patches upstream. Often upstream is also giving you a high quality review, that the SONiC developers/maintainers are not capable of.
> 
> Also the devicetree has a lot of commented out lines, that also need to be cleaned up.

Sorry for the ONL reference. I meant to say Linux. Anyway, I have cleaned up the patch to include file that was already committed instead creating new.. 
Why I don't want to upstream to Linux repo,
1) As you can see this patch file depends on many other patches. The other patch files all are in SONiC  repo and not in main Linux repo. So they cant apply my patch as it is invalid in that repo.
2) Our patch is applicable only for our switch platform running SONiC. So it applicable only for SONiC. It has no value in main Linux.

**Comment by prgeor** [comment]
@SelvaSupermicro please resolve the conflict

File: patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch
Comment (paulmenzel) [inline patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch:2]: Please add your real name and use one space.

    $ git config --global user.name "Selva Ra…"
    $ git commit --amend -s --author="Selva … <selvarajr@supermicro.com>"

File: patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch
Comment (paulmenzel) [inline patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch:8]: How is it changed?

File: patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch
Comment (paulmenzel) [inline patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch:136]: Is the leading / intended?

**Review by paulmenzel — CHANGES_REQUESTED** [review]

File: patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch
Comment (ErichYen) [reply patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch:8]: We use different SPI flash partitions from what is used in Marvell RD-AC5X dts file.
Supermicro SSE-G3748 smc_sse-g3748.dts uses:
                partition@0 {
                        label = "spi_flash_part0";
                        reg = <0x0 0x200000>;
                };

                parition@1 {
                        label = "spi_flash_part1";
                        reg = <0x200000 0x10000>;
                };

                parition@2 {
                        label = "spi_flash_part2";
                        reg = <0x210000 0xC00000>;
                };

                parition@3 {
                        label = "spi_flash_part3";
                        reg = <0xE10000 0x1F0000>;
                };

Marvell RD-AC5X ac5-98dx35xx-rd.dts uses:
                partition@0 {
                        label = "spi_flash_part0";
                        reg = <0x0 0x800000>;
                };

                parition@1 {
                        label = "spi_flash_part1";
                        reg = <0x800000 0x700000>;
                };

                parition@2 {
                        label = "spi_flash_part2";
                        reg = <0xF00000 0x100000>;
                };

File: patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch
Comment (ErichYen) [reply patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch:136]: Yes.  The leading / is intended.

**Comment by saiarcot895** [comment]
Sorry for the delay here, could you rebase your changes on top of the master branch, or merge in the current master branch into your branch? This is to resolve the patch conflict in `patch/series`.

**Comment by SelvaSupermicro** [comment]
> Sorry for the delay here, could you rebase your changes on top of the master branch, or merge in the current master branch into your branch? This is to resolve the patch conflict in `patch/series`.

Hi Saikrishna,
Thanks for the comment. Now, I have updated patch so that it can be merged with the latest master. 
Let me know if that is all good!

File: patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch
Comment (paulmenzel) [reply patch/0012-arm64-dts-marvell-Add-Supermicro-SSE-G3748-board.patch]: Why?

**Comment by paulmenzel** [comment]
Please improve the merge/pull request summary/title.

**Comment by ErichYen** [comment]
Hello SaiKrishna,
We may need your approve on @SelvaSupermicro new update here? 
https://github.com/sonic-net/sonic-linux-kernel/pull/373/commits/4aadc9131e6260c2c200b9be5f528a8b41bdb901

**Comment by saiarcot895** [comment]
Looks like there's one conflict somehow?

**Comment by saiarcot895** [comment]
> Please improve the merge/pull request summary/title.

Echoing this, particularly the title.

File: patch/series
Comment (saiarcot895) [inline patch/series:173]: There's too many changes here given that only one patch file being added in this PR. Please recheck the merge conflict resolution

**Pull request closed without merging**
